extensions/wikia/ContentReview/ContentReviewImportJS.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Wikia\ContentReview;
+
+class ImportJS {
+	const IMPORT_SCRIPTS = 'ImportJS';
+	const IMPORT_SCRIPTS_FUNCTION = 'importWikiaScriptPages';
+	const IMPORT_SCRIPTS_KEY = 'content-review-importjs';
+	const IMPORT_SCRIPTS_VERSION = '1.0';
+
+	/**
+	 * Get js script to load imports
+	 *
+	 * @return String
+	 */
+	public function getImportScripts() {
+		$importScript = \WikiaDataAccess::cache(
+			$this->getImportJSMemcKey(),
+			2592000, // 30 days,
+			function() {
+				$importScript = '';
+				$title = \Title::newFromText( self::IMPORT_SCRIPTS, NS_MEDIAWIKI );
+
+				if ( $title instanceof \Title && $title->getArticleID() != 0 ) {
+					$revision = \Revision::newFromTitle( $title );
+					$scripts = explode( PHP_EOL, $revision->getRawText() );
+
+					$imports = $this->prepareImports( $scripts );
+					$importScript = $this->createInlineScript( $imports );
+				}
+
+				return $importScript;
+			}
+		);
+
+		return $importScript;
+	}
+
+	/**
+	 * Check and filter provided scripts.
+	 *
+	 * Names should end with '.js' extension.
+	 * External scripts are allowed only from dev.wikia.com
+	 *
+	 * @param array $scripts
+	 * @return array
+	 */
+	public function prepareImports( Array $scripts ) {
+		$imports = [];
+
+		foreach( $scripts as $key => $script ) {
+			$script = trim( $script );
+
+			if ( strtoupper( substr( $script, -3 ) ) === '.JS' ) {
```

Comment:
In this case, you probably don't want to change the case when checking if it's a valid JS page. For importing JS pages, they should only end in `.js` lowercase. Pages ending in `.Js` or `.JS` aren't consider valid JS pages by MediaWiki, so `substr( $script, -3 ) === ContentReviewHelper::JS_FILE_EXTENSION` should be enough.